pkg/watcherdecisionengine/statefulset.go

```diff
@@ -0,0 +1,182 @@
+package watcherdecisionengine
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/affinity"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+	watcherv1beta1 "github.com/openstack-k8s-operators/watcher-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/watcher-operator/pkg/watcher"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"path/filepath"
+)
+
+const (
+	// KollaServiceCommand -
+	KollaServiceCommand = "/usr/local/bin/kolla_start"
+
+	// ComponentName -
+	ComponentName = watcher.ServiceName + "-decision-engine"
+)
+
+func StatefulSet(
+	instance *watcherv1beta1.WatcherDecisionEngine,
+	configHash string,
+	prometheusCaCertSecret map[string]string,
+	labels map[string]string,
+) *appsv1.StatefulSet {
+
+	// This allows the pod to start up slowly. The pod will only be killed
+	// if it does not succeed a probe in 60 seconds.
+	startupProbe := &corev1.Probe{
+		FailureThreshold: 6,
+		PeriodSeconds:    10,
+	}
+	// After the first successful startupProbe, livenessProbe takes over
+	livenessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds: 30,
+		PeriodSeconds:  30,
+	}
+	readinessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds: 30,
+		PeriodSeconds:  30,
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+	args := []string{"-c", KollaServiceCommand}
+
+	startupProbe.Exec = &corev1.ExecAction{
+		Command: []string{
+			"/usr/bin/pgrep", "-r", "DRST", ComponentName,
```

Comment:
we need -f too for comand name longer that 15 characters

Reply:
ack thanks doing as discussed in in slack just now